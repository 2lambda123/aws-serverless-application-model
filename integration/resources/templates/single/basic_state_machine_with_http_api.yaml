Resources:
  TestFunction:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: "exports.handler = async(event, context) => { console.log('te1st'); };"
      Handler: app.handler
      Runtime: nodejs12.x
      MemorySize: 128

  APIGatewayDeployment:
    Type: AWS::Serverless::HttpApi
    Properties:
      StageName: 'dev'
    DefinitionBody:
        info:
          title:
            Ref: AWS::StackName
          version: "1.0"
        openapi: 3.0.1

  MyStateMachine:
    Type: AWS::Serverless::StateMachine
    Properties:
      DefinitionUri: ${statemachineurl}
      Events:
        NormalPost:
          Type: HttpApi
          Properties:
            Path: /normal
            Method: post
            ApiId: 
              Ref: APIGatewayDeployment  
        StopPost:
          Type: HttpApi
          Properties:
            Path: /stop
            Method: post
            Action: stop
            ApiId: 
              Ref: APIGatewayDeployment  
        SyncPost:
          Type: HttpApi
          Properties:
            Path: /sync
            Method: post
            Action: startSync
            ApiId: 
              Ref: APIGatewayDeployment          
      Policies: 
        - LambdaInvokePolicy:
            FunctionName: 
              Ref: TestFunction