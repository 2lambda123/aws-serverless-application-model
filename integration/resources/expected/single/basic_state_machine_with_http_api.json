[
    { "LogicalResourceId":"APIGatewayDeploymentdevStage", "ResourceType":"AWS::ApiGatewayV2::Stage"},
    { "LogicalResourceId":"APIGatewayDeployment", "ResourceType":"AWS::ApiGatewayV2::Api" },
    { "LogicalResourceId":"APIGatewayDeploymentMyStateMachineRole", "ResourceType":"AWS::IAM::Role" },
    { "LogicalResourceId":"MyStateMachineRole", "ResourceType":"AWS::IAM::Role" },
    { "LogicalResourceId":"MyStateMachine", "ResourceType":"AWS::StepFunctions::StateMachine" },
    { "LogicalResourceId":"TestFunctionRole", "ResourceType":"AWS::IAM::Role" },
    { "LogicalResourceId":"TestFunction", "ResourceType":"AWS::Lambda::Function" }
] 